### PR TITLE
Add suggestions when search result not found

### PR DESCRIPTION
**Pull request by kshefchek**
Adds suggestions from SciGraph when a search result is not found.  This is a small enhancement while a more robust search/autocomplete widget is built with solr.

See #1196 

suggest

### CODE REVIEW

File: lib/monarch/web/webapp.js [hunk 1]
```diff
@@ -1076,10 +1076,24 @@ function searchHandler(request, term, fmt) {
...
             info.resultsTable = function() {return genTableOfSearchResults(info.results); };
             info.description = "<span>Results for "+term+ " searching phenotypes, diseases, genes, and models</span>";
         } else {
             info.resultsTable = ''; // "<span class=\"no-results\">&nbsp;&nbsp;No results found</span>";
+            info.hasResults = false;
+            var suggestions = engine.getSciGraphSuggestions(term, 5);
+            var suggestHead = "<h4>No results found, did you mean:</h4>";
+            var suggestSpan = '';
+            suggestions.forEach( function (i) {
+                var href = "<a href=\"/search/" + i + "\">" + i + "</a>";
```
Comment (DoctorBud) [inline lib/monarch/web/webapp.js]: style/readability: When writing suggestions.forEach, I think it is more understandable if the loop variable is named the singular of the collection variable. So in this case `i` should really be `suggestion`.

Comment (DoctorBud) [inline lib/monarch/web/webapp.js]: Do we know that `i` will always be in a URL format? Should it be URIEncoded prior to using it to build the `/search/...` url?

**Comment by DoctorBud** [comment]
Looks OK, provided that the URIEncoding issue above is considered. Provisional +1

File: lib/monarch/web/webapp.js (hunk 1 above)
Comment (kshefchek) [inline lib/monarch/web/webapp.js]: Good point, using function (i) is a bad habit, especially since it's not an index variable.

**Pull request merged**